File: backend/src/wattsup/schemas/status.py
from datetime import datetime

from pydantic import BaseModel, Field


class HiddenMetrics(BaseModel):
    output_voltage: bool = False
    input_frequency: bool = False


class UpsStatus(BaseModel):
    connected: bool
    ups_name: str
    last_poll_at: datetime
    error: str | None = None
    status: str | None = None
    battery_charge: float | None = None
    battery_voltage: float | None = None
    runtime_seconds: float | None = None
    load_percent: float | None = None
    input_voltage: float | None = None
    output_voltage: float | None = None
    input_frequency: float | None = None
    battery_date: str | None = None
    battery_test_result: str | None = None
    model: str | None = None
    manufacturer: str | None = None
    driver: str | None = None
    power_restored: bool = False
    hidden_metrics: HiddenMetrics = Field(default_factory=HiddenMetrics)

File: frontend/src/types/status.ts
export interface UpsStatus {
  connected: boolean;
  ups_name: string;
  last_poll_at: string;
  error: string | null;
  status: string | null;
  battery_charge: number | null;
  battery_voltage: number | null;
  runtime_seconds: number | null;
  load_percent: number | null;
  input_voltage: number | null;
  output_voltage: number | null;
  input_frequency: number | null;
  power_watts: number | null;
  power_source: "measured" | "estimated" | null;
  battery_date: string | null;
  battery_test_result: string | null;
  model: string | null;
  manufacturer: string | null;
  driver: string | null;
  power_restored: boolean;
  hidden_metrics: {
    output_voltage: boolean;
    input_frequency: boolean;
  };
}
